core/runtime/src/main/java/io/quarkus/runtime/configuration/DeploymentProfileConfigSource.java

```diff
@@ -12,8 +11,17 @@
  * A configuration source which supports deployment profiles.
  */
 public class DeploymentProfileConfigSource extends AbstractDelegatingConfigSource {
```

Comment:
We need all the property names to be available, so they can be copied into the defaults file, otherwise it is not possible to modify the profile at runtime. An extra hash map allocation here is not going to make any difference to anything performance wise.